src/api/externalUser.ts

```diff
@@ -276,12 +276,14 @@ export class ExternalUser {
         const issuer = await createDidKeyIssuerAndStore(keyManager);
         const publicKey = await keyManager.getKey({ level: KeyManagerLevel.BROWSER_LOCAL_STORAGE });
 
+        const hasPageParam = callbackPath.includes('page=');
+
         const loginRequestPayload: LoginRequestPayload = {
             login: {
                 randomString: randomString(32),
                 origin: window.location.origin,
                 publicKey: publicKey,
-                callbackPath,
+                callbackPath: callbackPath + (hasPageParam ? '&' : '?'),
```

Comment:
wouldn't it be better not to manipulate the callbackPath? this will cause developers a lot of confusion

Reply:
ok i see what you are trying to do. I think we need to
- leave the callback path here unmodified
- when we create the redirect URL path in `DualWalletResponse` we will need to aggregate any query parameters e.g. `?payload=...&page=bankless`